File: Sherlock’s Speedy Chase/Sherlock’s Speedy Chase Answer.py
# Sherlock’s Speedy Chase

from math import ceil

k = int(input()) # speed (m/s)
n = int(input()) # number of buildings 
a = list(map(int, input().split())) # height of each building
i = 0 # variable (number of building)
h = 0 # his location's height
target = 0 # target height
total = 0 # total time

while i < n: # main loop

    target = a[i] # target height
 
    if target > h: # if target is higher than his location
        temp = (target - h) / k # time
        total += ceil(temp) # add to total
        h = target # update

    else: # if target is lower than or equal to his location 
        temp = (h - target) / k # time
        total += ceil(temp) # add to total
        h = target # update

    i += 1 # update

total += n + ceil(h/k) # add to total
print(total)

File: Sherlock’s Speedy Chase/Sherlock’s Speedy Chase Answer.cpp
#include <iostream>
#include <vector>

using namespace std;


int main() {
    int k, n; // speed (m/s), number of buildings
    cin >> k >> n; // input
    vector<int> a(n); // height of each building
    for (int i = 0; i < n; ++i) {
        cin >> a[i]; // input
    } // input

    int h = 0; // his location's height
    int total = 0; // total time

    for (int target : a) {
        int diff = abs(target - h); // distance
        int time = (diff + k - 1) / k; // time
        total += time; // add to total
        h = target; // update
    } // main loop

    total += n + (h + k - 1) / k; // add to total

    cout << total << endl; // output

    return 0;
}